address_book/cmd/server/main.go

```diff
@@ -1,24 +1,57 @@
 package main
 
 import (
+	"addressbook/internal/config"
+	models "addressbook/internal/model"
 	"addressbook/internal/pb"
 	"addressbook/internal/service"
 	"log"
 	"net"
-	"os"
 
 	"google.golang.org/grpc"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
 )
 
+func connectDB(cfg *config.Config) (*gorm.DB, error) {
+	db, err := gorm.Open(postgres.Open(cfg.DBConn), &gorm.Config{})
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func initializeDB(db *gorm.DB) error {
+	if isInit := db.Migrator().HasTable(&models.Contact{}); !isInit {
```

Comment:
The migration is pretty simple and does not cover all things we need; for example, we don't have any constraint defined here. Due to that, you will need to add extra checks in all the methods, and the complexity of the program will grow.

Reply:
In my opinion, it is more correct to add constraints on the db side. (Such details were not specified in the task)

Reply:
Spreading the responsibility over the different parts of code - is not a good solution, at least from the maintainability perspective. It's better to make all DB-related operations in the same place, to keep the code support clean and understandable. It does not matter (for that little task, I mean) where the DB operations will be defined and done, but it should be the only place. And definitely, it's a bad idea to add some things "manually" on DB side, because it's the direct way to weird errors and non-reproducible issues.